primitives/message-dispatch/src/lib.rs

```diff
@@ -154,8 +163,15 @@ pub trait IntoDispatchOrigin<AccountId, Call, Origin> {
 
 /// A generic trait to validate message before dispatch.
 pub trait CallValidate<AccountId, Origin, Call> {
```

Comment:
I want to change these three generic params into associated types later. Then on `darwinia-common` side, we only maintain a general struct is enough. I need more tests locally.